packages/core/contracts/cross-chain-oracle/GovernorHub.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../common/implementation/Lockable.sol";
+import "./ParentMessengerInterface.sol";
+
+/**
+ * @title Governance relayer contract to be deployed on Ethereum that receives messages from the owner (Governor) and
+ * sends them to spoke contracts on child chains.
+ */
+contract GovernorHub is Ownable, Lockable {
+    // Associates chain ID with ParentMessenger contract to use to send governance actions to that chain's GovernorSpoke
+    // contract.
+    mapping(uint256 => ParentMessengerInterface) public messengers;
+
+    event RelayedGovernanceRequest(uint256 indexed chainId, address indexed messenger, address indexed to, bytes data);
+    event SetParentMessenger(uint256 indexed chainId, address indexed parentMessenger);
+
+    /**
+     * @notice Set new ParentMessenger contract for chainId.
+     * @param chainId child network that messenger contract will communicate with.
+     * @param messenger ParentMessenger contract that sends messages to ChildMessenger on network with ID `chainId`.
+     * @dev Only callable by the owner (presumably the Ethereum Governor contract).
+     */
+    function setMessenger(uint256 chainId, ParentMessengerInterface messenger) public nonReentrant() onlyOwner {
+        require(address(messenger) != address(0), "Invalid messenger contract");
```

Comment:
there could be an argument for setting this to er. this acts somewhat like a "delete" for a particular chain. consider if we wanted to remove a chain from operation, for example.

Reply:
You can also "delete" by setting the address to its own address, so I'll leave in for now.

Reply:
Actually I'll just remove this line for now as it doesn't make sense. It's just as easy to have a typo in the address that you pass to `setMessenger` as it is to accidentally call with a zero address. The owner can always reset the address